pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/MissingStaticMethodInNonInstantiatableClass.xml

```diff
@@ -380,6 +380,49 @@ public class Scratch {
         ]]></code>
     </test-code>
 
+    <test-code>
+
+        <description>#4225 [java] MissingStaticMethodInNonInstantiatableClass should consider Lombok's @NoArgsConstructor</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class Test {
+   private Test(int a, int b, int c, String abc, long d, double p,
+    String[] arr, int data, long in, float fl, String res) { } // 11 params
```

Comment:
I'll change these test samples, because the sample classes should have fields and not constructors. The constructors are generated by lombok, so no need to define them. But the generated constructors would depend on the fields (at least for allargs and required args). That way, the examples are more "realistic".